tests/common/normalizers/test_max_nesting.py

```diff
@@ -0,0 +1,79 @@
+from typing import List
+
+import dlt
+import pytest
+
+from dlt.common import json
+from tests.common.utils import json_case_path
+
+ALL_TABLES_FOR_RASA_EVENT = [
+    "bot_events",
+    "bot_events__metadata__known_recipients",
+    "bot_events__metadata__transaction_history__spend__target",
+    "bot_events__metadata__transaction_history__spend__starbucks",
+    "bot_events__metadata__transaction_history__spend__amazon",
+    "bot_events__metadata__transaction_history__deposit__employer",
+    "bot_events__metadata__transaction_history__deposit__interest",
+    "bot_events__metadata__vendor_list",
+]
+
+
+ALL_TABLES_FOR_RASA_EVENT_NESTING_LEVEL_2 = [
+    "bot_events",
+    "bot_events__metadata__known_recipients",
+    "bot_events__metadata__vendor_list",
+]
+
+
+@pytest.mark.parametrize(
+    "nesting_level,expected_num_tables,expected_table_names",
+    (
+        (0, 1, ["bot_events"]),
+        (1, 1, ["bot_events"]),
+        (2, 3, ALL_TABLES_FOR_RASA_EVENT_NESTING_LEVEL_2),
+        (5, 8, ALL_TABLES_FOR_RASA_EVENT),
+        (
+            15,
+            8,
+            ALL_TABLES_FOR_RASA_EVENT,
+        ),
+        (
+            25,
+            8,
+            ALL_TABLES_FOR_RASA_EVENT,
+        ),
+        (
+            1000,
+            8,
+            ALL_TABLES_FOR_RASA_EVENT,
+        ),
+    ),
+)
+def test_resource_max_nesting(
+    nesting_level: int, expected_num_tables: int, expected_table_names: List[str]
+):
+    @dlt.resource(max_table_nesting=nesting_level)
```

Comment:
you should have multiple resources with different nesting levels at the same time here, also add a source with a nesting level and check that it is overwritten by the resource setting but kept when there is no resource settingl.

Reply:
@sh-rp I updated the test with additional check 
1. setting the `max_table_nesting`
2. then running the pipeline
3. unsetting the `max_table_nesting`
4. check if the tables are still there.